packages/vue/src/components/basic/SelectedFilters.jsx

```diff
@@ -67,7 +71,43 @@ const SelectedFilters = {
 			this.$emit('clear', component, value);
 		},
 		clearValues() {
-			this.clearValuesAction();
+			const { resetToDefault, componentProps, selectedValues } = this;
+			if (resetToDefault) {
+				let valueToSet;
+				Object.keys(selectedValues).map(component => {
+					if (
+						!componentProps[component]
+						|| !componentProps[component].defaultValue
+						|| !componentProps[component].componentType
+					) {
+						return true;
+					}
+					if (
+						[
+							componentTypes.multiDropdownList,
+							componentTypes.multiList,
+							componentTypes.singleDropdownList,
+							componentTypes.singleList,
+							componentTypes.toggleButton,
+							componentTypes.multiRange,
+							componentTypes.dynamicRangeSlider,
+							componentTypes.singleRange,
+							componentTypes.dataSearch,
+							componentTypes.rangeSlider,
+							componentTypes.rangeInput,
+						].includes(componentProps[component].componentType)
+					) {
+						valueToSet = componentProps[component].defaultValue;
+					}
+
+					if (!isEqual(selectedValues[component].value, valueToSet)) {
+						this.setValue(component, valueToSet);
```

Comment:
@mohdashraf010897 Instead of calling `setValue` for each component you should create a single action to reset the values for all components. The current solution is not performance efficient and could lead to re-rendering issues.